granulate_utils/metadata/cloud.py

```diff
@@ -38,7 +38,14 @@ class AwsInstanceMetadata(InstanceMetadataBase):
 
 
 @dataclass
-class GcpInstanceMetadata(InstanceMetadataBase):
+class AwsContainerMetadata(CloudMetadataBase):
```

Comment:
This should have `provider: aws` as well.

Reply:
it is initialized with provider="aws". do you mean set as default `provider: str = 'aws'`?

Reply:
We have the provider field in `AwsInstanceMetadata` but not in `AwsContainerMetadata`

Reply:
<img width="538" alt="image" src="https://github.com/Granulate/granulate-utils/assets/1116272/917cad37-cb5b-486a-866b-f4ebce495d02">

we have the provider field everywhere. its not explicitly mentioned in AwsInstanceMetadata either, and it doesn't need to be. it is unfortunate that we have the inconsistency of it being mentioned in Gcp... and Azure... even when they inherit it from the base class as well.

Reply:
Oof, sorry, I looked at GCP and Azure and not AWS so it confused me. You can fix GCP and Azure if you'd like on this occasion.

Reply:
I haven't decided myself whether it's better for it to be mentioned and set to a default, or just inherited and not mentioned 🤔 
Perhaps we can revisit the question in the future